pkg/storage/tsdb/block/block.go

```diff
@@ -40,6 +40,12 @@ const (
 
 	// DebugMetas is a directory for debug meta files that happen in the past. Useful for debugging.
 	DebugMetas = "debug/metas"
+
+	// OutOfOrderExternalLabelKey is the external label used to mark blocks as containing out-of-order data
+	OutOfOrderExternalLabelKey = "__out_of_order__"
+
+	// OutOfOrderExternalLabelValue is the value to be used for the OutOfOrderExternalLabelKey label
+	OutOfOrderExternalLabelValue = "true"
```

Comment:
I feel like this should belong to the tsdb package (see [here](https://github.com/grafana/mimir/blob/ooo-blocks-optimization/pkg/storage/tsdb/config.go#L24)), but it was causing cyclic imports and didn't want to get into a refactor. Also, we don't really care about the value the label has, so maybe it's not worth this constant.

Thoughts on those 2 things?